app/models/reporter.go

```diff
@@ -0,0 +1,335 @@
+/*
+   Copyright (c) 2016, Percona LLC and/or its affiliates. All rights reserved.
+
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU Affero General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU Affero General Public License for more details.
+
+   You should have received a copy of the GNU Affero General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>
+*/
+
+package models
+
+import (
+	"bytes"
+	"log"
+	"text/template"
+	"time"
+
+	"github.com/percona/qan-api/app/db/mysql"
+)
+
+// report provide methods to works with query report
+type report struct{}
+
+// Report instance of report model
+var Report = report{}
+
+type QueryRank struct {
+	Rank        uint    // compared to global, same as Profile.Ranks index
+	Percentage  float64 // of global value
+	Id          string  // hex checksum
+	Abstract    string  // e.g. SELECT tbl
+	Fingerprint string  // e.g. SELECT tbl
+	QPS         float64 // ResponseTime.Cnt / Profile.TotalTime
+	Load        float64 // Query_time_sum / (Profile.End - Profile.Begin)
+	FirstSeen   time.Time
+	Log         []QueryLog
+	Stats       Stats // this query's Profile.Metric stats
+}
+
+type QueryLog struct {
+	Point          uint      `db:"Point"`
+	Start_ts       time.Time `db:"Start_ts"`
+	Query_count    float32   `db:"Query_count"`
```

Comment:
don't use underscores in Go names; struct field Query_count should be QueryCount